### ISSUE TITLE
ORA-00001 returned before ut_coverage_html_reporter finished

### ISSUE THREAD

**Issue by hehawe** [utPLSQL/utPLSQL#1086]
Hi,
When I run something like
utPLSQL-cli/bin/utplsql run xxx/yyy@//zzz:1521/localpdb -c -f=ut_coverage_html_reporter -o=coverage.html
I get the following error

Successfully connected to database. UtPLSQL core: v3.1.10.3349
Oracle-Version: 18.0.0.0.0
Running tests now.
--------------------------------------
TestRunner initialized
Running on utPLSQL v3.1.10.3349
Initializing reporters
Running tests
ORA-00001: unique constraint (UT3.UT_COVERAGE_SOURCES_TMP_PK) violated
ORA-06512: at "UT3.UT_RUNNER", line 180
ORA-06512: at "UT3.UT_COVERAGE_HELPER", line 37
ORA-06512: at "UT3.UT_COVERAGE", line 166
ORA-06512: at "UT3.UT_COVERAGE", line 235
ORA-06512: at "UT3.UT_COVERAGE_HTML_REPORTER", line 35
ORA-06512: at "UT3.UT_REPORTER_BASE", line 193
ORA-06512: at "UT3.UT_EVENT_MANAGER", line 70
ORA-06512: at "UT3.UT_EVENT_MANAGER", line 80
ORA-06512: at "UT3.UT_RUN", line 74
ORA-06512: at "UT3.UT_SUITE_ITEM", line 49
ORA-06512: at "UT3.UT_RUNNER", line 172
ORA-06512: at line 1
	at oracle.jdbc.driver.T4CTTIoer11.processError(T4CTTIoer11.java:509)
	at oracle.jdbc.driver.T4CTTIoer11.processError(T4CTTIoer11.java:461)
	at oracle.jdbc.driver.T4C8Oall.processError(T4C8Oall.java:1104)
	at oracle.jdbc.driver.T4CTTIfun.receive(T4CTTIfun.java:550)
	at oracle.jdbc.driver.T4CTTIfun.doRPC(T4CTTIfun.java:268)
	at oracle.jdbc.driver.T4C8Oall.doOALL(T4C8Oall.java:655)

Am I doing something wrong or there's a bug?
I get the same result running ut_coverage_html_reporter in sql developer, so it's not related just to cli.

**Comment by jgebal** [comment]
@hehawe 
You can run the following query to identify duplicate object names that are causing this problem in your test run.
```sql
select owner, object_name, min(object_type) first_object_type, max(object_type) last_object_type, count(1) duplicate_object_name_count
  from dba_objects
 where object_type in  ('PACAKGE BODY','TYPE BODY','PROCEDURE','FUNCTION','TRIGGER')
 group by owner, object_name
 having count(1) > 1
 order by owner, object_name
```


I've managed to reproduce the issue using the following steps:
```sql
create or replace package test_duplicate_name is
  procedure some_procedure;
end;

create or replace package body test_duplicate_name is
  procedure some_procedure is
  begin
    null;
  end;
end;
/

create table test_table (id integer);

create or replace trigger test_duplicate_name
  before insert on test_table
begin
  null;
end;

begin
  ut.run(ut_coverage_html_reporter());
end;
```

And I go the error as described:
```
ORA-00001: unique constraint (UT3.UT_COVERAGE_SOURCES_TMP_PK) violated
ORA-06512: at "UT3.UT_RUNNER", line 180
ORA-06512: at "UT3.UT_COVERAGE_HELPER", line 37
ORA-06512: at "UT3.UT_COVERAGE", line 166
ORA-06512: at "UT3.UT_COVERAGE", line 235
ORA-06512: at "UT3.UT_COVERAGE_HTML_REPORTER", line 35
ORA-06512: at "UT3.UT_REPORTER_BASE", line 193
ORA-06512: at "UT3.UT_EVENT_MANAGER", line 70
ORA-06512: at "UT3.UT_EVENT_MANAGER", line 80
ORA-06512: at "UT3.UT_RUN", line 74
ORA-06512: at "UT3.UT_SUITE_ITEM", line 49
ORA-06512: at "UT3.UT_RUNNER", line 172
ORA-06512: at "UT3.UT", line 134
ORA-06512: at "UT3.UT", line 488
ORA-06512: at "UT3.UT", line 558
ORA-06512: at line 2
```

## Workaround for now

Rename duplicate objects.
It should be easiest to rename triggers as they should have no code depending on them and so the impact of rename should be zero.

Long-term, a fix is needed in utPLSQL to address this issue.

Should be relatively easy to fix now that we know what the cause is.

**Comment by hehawe** [comment]
@jgebal 

Hi Jacek,
Thanks for the workaround. We have a really big project, and yes, I could identify the duplicates. And there's a progress. However still with no success at the end. Now I have the following error:

```
ORA-19011: Character string buffer too small
ORA-06512: at "UT3.UT_RUNNER", line 180
ORA-06512: at "SYS.DBMS_XMLGEN", line 19
ORA-06512: at "SYS.DBMS_XMLGEN", line 295
ORA-06512: at "UT3.UT_COVERAGE_REPORT_HTML_HELPER", line 190
ORA-06512: at "UT3.UT_COVERAGE_REPORT_HTML_HELPER", line 202
ORA-06512: at "UT3.UT_COVERAGE_REPORT_HTML_HELPER", line 334
ORA-06512: at "UT3.UT_COVERAGE_HTML_REPORTER", line 37
ORA-06512: at "UT3.UT_REPORTER_BASE", line 193
ORA-06512: at "UT3.UT_EVENT_MANAGER", line 70
ORA-06512: at "UT3.UT_EVENT_MANAGER", line 80
ORA-06512: at "UT3.UT_RUN", line 74
ORA-06512: at "UT3.UT_SUITE_ITEM", line 49
ORA-06512: at "UT3.UT_RUNNER", line 172
ORA-06512: at "UT3.UT", line 134
ORA-06512: at "UT3.UT", line 488
ORA-06512: at "UT3.UT", line 623
ORA-06512: at line 1
19011. 00000 -  "Character string buffer too small" 
*Cause:    The string result asked for is too big to return back
*Action:   Get the result as a lob instead
```

Thanks for your effort,
HHW

**Comment by jgebal** [comment]
Can you raise this as a separate issue please?
I will not be able to investigate it this week.
I'm on holidays

**Comment by jgebal** [comment]
@hehawe 
Your [other problem](https://github.com/utPLSQL/utPLSQL/issues/1086#issuecomment-652208837) was also identified now and will be fixed in version 3.1.13
